src/metabase/query_processor/util.clj

```diff
@@ -39,7 +39,12 @@
 (defmulti query->remark
   "Generate an appropriate remark `^String` to be prepended to a query to give DBAs additional information about the query
   being executed. See documentation for `mbql->native` and #2386.
-  for more information."
+  for more information.
+
+  Note that this messes up the visualization errors for native queries.
+  Therefore there is a very very long-distance dependency that this causes on the VisualizationErrors
+  for native queries.
+  "
```

Comment:
I'm not sure i understand this comment. Also it should be 118 characters wide if possible.

Reply:
we futz with the query in here: we do mutations that are intended to unfutz the errors for the query alllll the way in adjustPositions all the way in the frontend (the alternative I can think of is mutating the actual errors, which is scarier). I will edit comment tho, it is indeed confusing